### ISSUE TITLE
[Environment] Upgrade with latest stable before/once they're published

### ISSUE THREAD

**Issue by AlexV525** [dart-lang/pub-dev#5352]
This is not the first time I saw things like this, a new stable version of Dart/Flutter gets published, and pub.dev still using the previous stable version for analysis. If we have problems when upgrading, it should be announced, or any release should be held since pub.dev is the most important part of the whole ecosystem.

Current: Analysed with Pana `0.21.4`, Flutter `2.5.3`, Dart `2.14.4`. (Dec 13, 2021)

**Comment by sigurdm** [comment]
We cannot analyze with an sdk before it is released.

Re-analyzing all packages takes some time.

**Comment by AlexV525** [comment]
> We cannot analyze with an sdk before it is released.
> 
> Re-analyzing all packages takes some time.

Yeah, sure I understand there will be some duration between publish and analyze, but packages, taking mine for example, published after the new stable is released, yet still analyzed by an older sdk, which make it unidentified.

https://pub.dev/packages/wechat_assets_picker

**Comment by isoos** [comment]
Analysis report shows:
```
OUT:
Resolving dependencies...
ERR:
The current Flutter SDK version is 2.5.3.

Because wechat_assets_picker requires Flutter SDK version >=2.8.0, version solving failed.
```
We haven't released a new analysis runtime for Flutter SDK 2.8 yet (coming in the next few days), but it could have been picked up by the preview analysis SDKs. This may be a bug in the preview-detection?

**Comment by AlexV525** [comment]
As I can imagine, it'll be more reasonable that we start to analyze packages with new sdk once it served as distribution. Those packages already started or awating for analyze could still use the older one, and update after new queue is done.

**Comment by AlexV525** [comment]
> but it could have been picked up by the preview analysis SDKs. This may be a bug in the preview-detection?

Maybe, as I saw a document said pub will recognize releases with unstable SDKs as pre-releases.

**Comment by isoos** [comment]
> As I can imagine, it'll be more reasonable that we start to analyze packages with new sdk once it served as distribution. Those packages already started or awating for analyze could still use the older one, and update after new queue is done.
> Maybe, as I saw a document said pub will recognize releases with unstable SDKs as pre-releases.

We have several, somewhat contradicting features here:
- We want to have a stable and bug-free analysis provided to packages. (takes time to run staging and real-world tests, and we are not too eager to release new site versions on a whim...)
- We want to follow the latest SDKs as soon as possible. (could introduce bugs)
- We already have two set of SDKs (stable and prerelease), and it seems that this specific package should have been picked up by the prerelease.

Please bear with us while we figure out what's went wrong in this case, we will try to deploy new SDKs and also fixes on a tight schedule.

**Comment by isoos** [comment]
This has been fixed, and a new site was released.

**Comment by AlexV525** [comment]
> We cannot analyze with an sdk before it is released.
> 
> Re-analyzing all packages takes some time.

@sigurdm @isoos Can we put the state of packages into pending analysis rarher than some background non-notified tasks? Scores are always corrupted after new minor version stable released.

**Comment by sigurdm** [comment]
We have recently launched a new analysis-worker system, that makes analysing all packages much faster. Hopefully that should make it quicker to bring analysis with new releases.

> Can we put the state of packages into pending analysis rarher than some background non-notified tasks?

Not sure what this means.

> Scores are always corrupted after new minor version stable released.

Can you explain what you mean by "always corrupted"

cc @jonasfj

**Comment by jonasfj** [comment]
I think our new analysis system has some issues with falling back to old results, we're working to address that. In any case, that would explain the partially broken scores yesterday.

**Comment by AlexV525** [comment]
We published packages to support the latest Flutter release (because of breaking changes). By the time we published, it always shows UNKNOWN PLATFORM.

**Comment by AlexV525** [comment]
As far as I look into, we manually bump runtime versions. Could we do that by fetching the latest version for channels from the storage API and roll automatically?

**Comment by isoos** [comment]
> Could we do that by fetching the latest version for channels from the storage API and roll automatically?

We may do that eventually, but have a few other items on our agenda to roll out before we can do that safely.

**Comment by AlexV525** [comment]
Thanks. I'd schedule it more agressively since things need pushes. How can we (outside of the team) determine what remains to do?

I'll submit a new issue for the proposal above.
